xbmc/platform/darwin/ios-common/AnnounceReceiver.mm

```diff
@@ -93,10 +94,52 @@ id objectFromVariant(const CVariant &data)
 
 void AnnounceBridge(ANNOUNCEMENT::AnnouncementFlag flag, const char *sender, const char *message, const CVariant &data)
 {
+  int item_id = -1;
+  std::string item_type = "";
+  CVariant nonConstData = data;
+  const std::string msg(message);
+
+  // handle data which only has a database id and not the metadata inside
+  if (msg == "OnPlay")
+  {
+    if (!nonConstData["item"].isNull())
+    {
+      if (!nonConstData["item"]["id"].isNull())
+      {
+        item_id = (int)nonConstData["item"]["id"].asInteger();
+      }
+
+      if (!nonConstData["item"]["type"].isNull())
+      {
+        item_type = nonConstData["item"]["type"].asString();
+      }
+    }
+  
+    // if we got an id from the passed data
+    // we need to get title, track, album and artist from the db
+    if (item_id >= 0)
+    {
+      if (item_type == MediaTypeSong)
+      {
+        CMusicDatabase db;
+        if (db.Open())
+        {
+          CSong song;
+          if (db.GetSong(item_id, song))
+          {
+            nonConstData["item"]["title"] = song.strTitle;
+            nonConstData["item"]["track"] = song.iTrack;
+            nonConstData["item"]["album"] = song.strAlbum;
+            nonConstData["item"]["artist"] = song.GetArtist();
```

Comment:
I need the array here - thats what is in there when we don't have the item scanned into the db - the rest of the code relies on it. I simply try to mimic the data as if it was announced from a non scanned item (where all that metadata is already filled in the data param) - its only when the item is scanned into the db that the announce caller only puts the db id into that data dictionary...